File: kattis/easy/java/conundrum.java
package easy;

import java.util.Arrays;
import java.util.Objects;
import java.util.Scanner;

public class conundrum {
	public static void main(String[] args) {
		try (Scanner scanner = new Scanner(System.in)) {
			String[] line = scanner.nextLine().split("");

			String[] per = {"P", "E", "R"};
			int index = 0;
			int days = 0;

			// changes letters successively, one letter per day
			for (String s : line) {
				if (index > 2) index = 0; 				// resets letter-index
				if (!Objects.equals(s, per[index]))
					days++; index++;					// days and current index increases
			}

			System.out.println(days);
		}
	}
}

File: kattis/easy/cpp/conundrum.cpp
//
// Created by BBJ on 05.01.2024.
//
#include <iostream>
#include <string>

int main() {
    std::string line;
    std::getline(std::cin, line);

    char per[] = {'P', 'E', 'R'};
    int index = 0;
    int days = 0;

    // change letters successively, one letter per day
    for (const char& c : line) {
        if (index > 2) index = 0;
        if (c != per[index])
            days++;
        index++;
    }

    std::cout << days; // print days needed to convert full line to (PER)^*

    return 0;
}